client/ayon_core/hosts/max/plugins/publish/validate_instance_in_context.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+"""Validate if instance asset is the same as context asset."""
+import pyblish.api
+from ayon_core.pipeline.publish import (
+    RepairAction,
+    ValidateContentsOrder,
+    PublishValidationError,
+    OptionalPyblishPluginMixin
+)
+from ayon_core.hosts.max.api.action import SelectInvalidAction
+from pymxs import runtime as rt
+
+
+class ValidateInstanceInContext(pyblish.api.InstancePlugin,
+                                OptionalPyblishPluginMixin):
+    """Validator to check if instance asset match context asset.
+
+    When working in per-shot style you always publish data in context of
+    current asset (shot). This validator checks if this is so. It is optional
+    so it can be disabled when needed.
+
+    Action on this validator will select invalid instances.
+    """
+    order = ValidateContentsOrder
+    label = "Instance in same Context"
+    optional = True
+    hosts = ["max"]
+    actions = [SelectInvalidAction, RepairAction]
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+
+        folderPath = instance.data.get("folderPath")
+        task = instance.data.get("task")
+        context = self.get_context(instance)
+        if (folderPath, task) != context:
+            context_label = "{} > {}".format(*context)
+            instance_label = "{} > {}".format(folderPath, task)
+            message = (
+                "Instance '{}' publishes to different folderPath than current "
```

Comment:
Folder path might be confusing here since it's also potentially publishing to a different task?
```suggestion
                "Instance '{}' publishes to different context than current "
```